compile/compile.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package compile
+
+import (
+	"github.com/mohae/deepcopy"
+
+	taxonomyio "fybrik.io/fybrik/pkg/taxonomy/io"
+	"fybrik.io/fybrik/pkg/taxonomy/model"
+)
+
+// Files generates a taxonomy document from a base file and zero or more layer files
+func Files(baseDocPath string, layerDocPaths []string, opts ...Option) (*model.Document, error) {
+	// load base document
+	base, err := taxonomyio.ReadDocumentFromFile(baseDocPath)
```

Comment:
the methods from `taxonomy/io` are used only here, instead of include them from fybrik you can just copy (move later) them to here